api/observability/v1/output_types.go

```diff
@@ -411,8 +401,20 @@ type Elasticsearch struct {
 	// +kubebuilder:validation:Optional
 	Tuning *ElasticsearchTuningSpec `json:"tuning,omitempty"`
 
-	// defaults to: log_type-write
-	IndexSpec `json:",inline"`
+	// Index is the index for the logs. This supports template syntax
+	// to allow dynamic per-event values.
+	//
+	// The Index can be a combination of static and dynamic values consisting of field paths followed by `||` followed by another field path or a static value.
+	// A dynamic value is encased in single curly brackets `{}` and MUST end with a static fallback value separated with `||`.
+	// Static values can only contain alphanumeric characters along with dashes, underscores, dots and forward slashes.
+	// Example:
+	// 1. foo-{.bar||"none"}
+	// 2. {.foo||.bar||"missing"}
+	// 3. foo.{.bar.baz||.qux.quux.corge||.grault||"nil"}-waldo.fred{.plugh||"none"}
+	//
+	// +kubebuilder:validation:Pattern:=`^(([a-zA-Z0-9-_.\/])*(\{(\.[a-zA-Z0-9_]+|\."[^"]+")+((\|\|)(\.[a-zA-Z0-9_]+|\.?"[^"]+")+)*\|\|"[^"]*"\})*)*$`
+	// +kubebuilder:validation:Required
```

Comment:
I believe this is redundant give we have defined a pattern that does not allow spaces